webpack/webpack5-module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type {
+  Chunk,
+  Compilation,
+  Compiler,
+  Module,
+  ModuleGraph,
+  WebpackPluginInstance,
+  sources
+} from 'webpack';
+
+import { Template, dependencies } from 'webpack';
```

Comment:
You can fix this import by defining the classes inside of `apply`; that's what I did in another recent plugin with custom Dependency classes.